src/anonymous-credentials/presentation-builder.ts

```diff
@@ -242,11 +242,13 @@ export class PresentationBuilder extends Versioned {
   enforceBounds(
     credIdx: number,
     attributeName: string,
-    min: number,
-    max: number,
+    vmin: number | Date,
+    vmax: number | Date,
     provingKeyId: string,
     provingKey?: LegoProvingKey | LegoProvingKeyUncompressed
   ) {
+    const min = vmin instanceof Date ? vmin.getTime() : vmin;
```

Comment:
The conversion of date to integer should be done with the encoder and not duplicated here.

Reply:
what has this todo with the encoder? this is simplyifying the APi for the user so they can pass date objects instead of knowing to call getTime for enforceBounds

Reply:
I didn't say call `getTime` for the args. You're calling `getTime` directly here exposing encoder details when you already have access to the encoder inside this function and can call it on min, max. If in the future any change is made to the encode function, that change has to be made here as well.

Reply:
are you referring to `Encoder.dateEncoder`? that returns a function which returns an encoded message, rather than numeric value - so i dont think i want to use that directly. i could extract the string/date obj parsing into a utility called by both instead?

Reply:
Yes, I mean the function that converts to a number

Reply:
makes sense now, thought you were referring to the actual encoding somehow and thought i misunderstood the code path from enforceBounds. will do that